runtime/lua/vim/fs.lua

```diff
@@ -368,7 +371,7 @@ function M.root(source, marker)
 
   local paths = M.find(marker, {
     upward = true,
-    path = path,
+    path = vim.fn.fnamemodify(path, ':p:h'),
```

Comment:
Can't hurt to document this (that a string `{source}` can both be relative and absolute).

Same for file vs. directory (if both are supported now).

Reply:
7f140398b9ec028a5396777d29c1e7f25d262ebe